openlibrary/catalog/add_book/__init__.py

```diff
@@ -522,7 +525,7 @@ def find_exact_match(rec, edition_pool):
     return False
 
 
-def find_match(e1, edition_pool):
+def find_marc_match(e1, edition_pool):
```

Comment:
@scottbarnes This should be renamed, there isn't anything here that is a proper MARC record.

The source of the `marc` name is `build_marc()`
https://github.com/internetarchive/openlibrary/blob/ccb37150a1a362e47a374caa8cb1efdf52fca30e/openlibrary/catalog/merge/merge_marc.py#L314

but that's not a MARC either, it's an "expanded representation of an edition dict" ... that needs a more descriptive name too.

The whole "merge" terminology of https://github.com/internetarchive/openlibrary/tree/ccb37150a1a362e47a374caa8cb1efdf52fca30e/openlibrary/catalog/merge  is confusing (OL uses "merge" to mean merging authors, works, and editions through the UI. This use of "merge" on import seems to be more around finding matches.

This method is about finding an existing OL record match, `find_exisiting_match()` or maybe the original `find_match()` is sufficient?  `marc` just isn't accurate though.

I agree `find_quick_match()` is an improvement over the original `early_exit()` though!

Reply:
I've looked at the rest of the code and I think a way forward would be to call this method `find_enriched_match()` (using your terminology from `find_match()`, which is good in that it collects `find_quick_match()`, `find_exact_match()`, and this other one.

Perhaps move 
```
        enriched_rec = build_marc(rec)
        add_db_name(enriched_rec)
```
into this `find_enriched_match()` method so it "enriches" the record and attempts to match it too?

Reply:
`build_marc()` could be called `expand_record()`, and moving it out of the `merge` area might be an idea because it looks like most of the other code there was created for some other purpose

Reply:
@hornc, thank you very much for this feedback.

I've implemented all your suggestions, save for the last one about renaming `build_marc()` and relocating it.

I agree that should be done, but as I was in the process of carrying this out, I noticed there is significant overlap between the contents `merge/merge.py` and `merge/merge_marc.py`, to the point a number of identical, non-deprecated functions appear in both (and some have very slight differences). E.g. `build_titles()` appears in both, that's a function upon which what is now called `build_marc()` depends.

I was thinking it might be worth tackling that consolidation (and function relocation, and possible deprecation) in a separate PR. If you agree I can create an issue and have a go at trying to do that.

Reply:
@scottbarnes thanks! That sounds perfectly reasonable to do it in another PR. I know how these improvements can get out of hand. There are, as you note, other issues that could be improved at the same time, so doing in a different PR / issue is a good idea!